backend/src/services/exportService.ts

```diff
@@ -0,0 +1,150 @@
+/**
+ * Functionalities required by the export router (/api/export/).
+ *
+ * Offers functions for handling data and saving it between requests during the export procedure.
+ * The data is saved in dataObjects array in server's main memory (RAM).
+ * Server storage is not used.
+ */
+
+import xlsx from 'xlsx';
+import { getRndString } from './getRndString';
+
+interface Box {
+    id: string;
+    input: string;
+    output: string;
+}
+
+interface Panel {
+    category: string;
+    panels: Panel[];
+    boxes: Box[];
+}
+
+interface ExportData {
+    theme: string;
+    panels: Panel[];
+}
+
+interface DataObject {
+    id: string;
+    fileName: string;
+    data: Buffer;
+}
+
+//The IDs should be long enought to prevent random guessing
+const idLenght = 30;
+
+//The max number of data objects is limited to limit memory consumption
+//However, this still doesn't limit the size of an individual data object
+const maxDataObjects = 50;
+const dataObjects: DataObject[] = [];
+
+/**
+ * Appends a new data object to dataObjects.
+ * Removes the oldest object in the array if max length would be exceeded.
+ */
+const appendDataObject = (fileName: string, data: Buffer): string => {
+    if (dataObjects.length >= maxDataObjects) dataObjects.shift();
+    const id = getRndString(idLenght);
+    dataObjects.push({ id, fileName, data });
+    return id;
+};
+
+/**
+ * Reads and removes a data object with corresponding ID from dataObjects.
+ * The returned fileName and data will both be null if no data object was found.
+ */
+const readDataObject = (
+    id: string
+): { fileName: string | null; data: Buffer | null } => {
+    const i = dataObjects.findIndex((o) => o.id === id);
+    if (i === -1) return { fileName: null, data: null };
+    const { fileName, data } = dataObjects[i];
+    dataObjects.splice(i, 1);
+    return { fileName, data };
+};
+
+/**
+ * Recursively calculate the max depth of panel tree.
+ * The returned number tells how many category columns will be required.
+ */
+const getMaxDepth = (panels: Panel[]): number => {
+    let max = 0;
+    for (const panel of panels) {
+        const depth = 1 + getMaxDepth(panel.panels);
+        if (depth > max) max = depth;
+    }
+    return max;
+};
+
+/**
+ * Writes the column headers to an aoa.
+ * The aoa is an array of arrays that represents an excel work sheet.
```

Comment:
describing parameters or constants in the docstring can be more elegantly performed with
@param {type} name description
or
@constant {type} name description